cmd/up/exec.go

```diff
@@ -233,9 +241,11 @@ func (eg *envsGetter) getEnvs(ctx context.Context) ([]string, error) {
 	}
 	envs = append(envs, configMapEnvs...)
 
-	for _, env := range apps.GetDevContainer(app.PodSpec(), "").Env {
-		envs = append(envs, fmt.Sprintf("%s=%s", env.Name, env.Value))
+	devContainerEnvs, err := eg.devContainerEnvGetter.getEnvsFromDevContainer(ctx, app.PodSpec(), "", eg.namespace, eg.client)
+	if err != nil {
+		return nil, err
 	}
+	envs = append(envs, devContainerEnvs...)
 
 	envs = append(envs, eg.getDefaultLocalEnvs()...)
```

Comment:
```suggestion
	envs = append(envs, devContainerEnvs..., eg.getDefaultLocalEnvs()...)
```

maybe append both at the same operation?

Reply:
Yes, This code could be optimised by retrieving all the env vars in parallel and then adding them by the prioritisation  order. But I don't think this should be covered in this PR